### PR TITLE
Neat log v2

### PR DESCRIPTION
**Pull request by shirazbe**
updated log

### CODE REVIEW

File: genetic.py [hunk 1]
```diff
@@ -2,35 +2,45 @@
 from network_components import *
 from graph import Graph
 from util import safe_softmax
-
+import pandas as pd
+from prettytable import PrettyTable
+DEBUG = False
```
Comment (tomjur) [inline genetic.py]: general question:
the DEBUG is used to print the text file that you prefer to work with (over the csv) - right?
[no changes needed]

File: genetic.py [hunk 2]
```diff
@@ -96,10 +108,68 @@ def __init__(self, config):
...
+class InnovationTrackerLog(InnovationLog):
+    def __init__(self, filename):
+        InnovationLog.__init__(self, filename)
+
+
+class InnovationSampleLog(InnovationLog):
+    def __init__(self, filename):
+        InnovationLog.__init__(self, filename)
+        self.w = None
+
```
Comment (tomjur) [inline genetic.py]: where are you using w?
if not used both logs can be instances of InnovationLog which could make the code simpler

File: genetic.py [hunk 3]
```diff
@@ -243,7 +321,11 @@ def _crossover(self, innovation_history1, fitness_score1, innovation_history2, f
                 child_innovation_history.append((innovation_number2, structure))
             # increase index2
             index2 += 1
-        return child_innovation_history
+
+        child_innovation_sample = Sample(child_innovation_history)
+        self.log.sample_id = child_innovation_sample.my_id
+        self.instanceLog.sample_id = child_innovation_sample.my_id
+        return child_innovation_sample
```
Comment (tomjur) [inline genetic.py]: i thought that we removed the sample id from the log, right?

File: genetic.py [hunk 4]
```diff
@@ -318,21 +400,26 @@ def _get_innovation_history_instance_copy(innovation_history_instance):
             result.append(new_tuple)
         return result
 
-    def _mutate(self, innovation_history_instance, existing_added_neurons):
+    def _mutate(self, sample, existing_added_neurons):
+        # sample might evolve now => keep track of which sample we are looking at
+        self.log.sample_id = sample.my_id
+        self.instanceLog.sample_id = sample.my_id
+
```
Comment (tomjur) [inline genetic.py]: same

File: genetic.py [hunk 5]
```diff
@@ -418,27 +505,32 @@ def _mutate_add_connection(self, innovation_history_instance):
...
                     # add to innovation tracker
                     new_connection, new_connection_index = self.innovation_tracker.add_connection(
-                        source_id, destination_id)
+                        source_id, destination_id, sample_id)
+
                 # create a new connection instance with 0 weight
                 new_connection_instance = ConnectionInstance(new_connection, 0.0)
+
+                self.instanceLog.add_connection_log(new_connection_index, source_id, destination_id, sample_id)
+                # , new_connection_instance.weight)
```
Comment (tomjur) [inline genetic.py]: please remove comment

File: genetic.py (hunk 1 above)
Comment (shirazbe) [reply genetic.py]: right.

File: genetic.py (hunk 2 above)
Comment (shirazbe) [reply genetic.py]: deleted w. 
now only InnovationLog class exists.

File: genetic.py (hunk 3 above)
Comment (shirazbe) [reply genetic.py]: right, I missed that. deleted now.

File: genetic.py (hunk 4 above)
Comment (shirazbe) [reply genetic.py]: fixed

File: genetic.py (hunk 5 above)
Comment (shirazbe) [reply genetic.py]: fixed

**Pull request merged**